### PR TITLE
feat: support filters for opposite-news endpoint

### PR DESCRIPTION
**Pull request by RaymondHuang210129**

### CODE REVIEW

File: app.py [hunk 1]
```diff
@@ -283,9 +297,64 @@ def analyze_bias(
...
     params: SearchParam,
     count: int,
-    call_api: Callable[[SearchParam, int], Union[SearchSuccess, SearchError]],
+    call_api: Callable[[SearchParam, int, bool], Union[SearchSuccess, SearchError]],
 ) -> Union[SearchSuccess, SearchError]:
-    return call_api(params, count)
+    return call_api(params, count, True)
+
+
+def search_news_with_filter(
```
Comment (seanwu1105) [inline app.py:305]: Not sure the reason to have these two functions separately. Merging them with optional parameters seems to be a good idea for me.

File: app.py [hunk 2]
```diff
@@ -103,11 +108,12 @@ def generate_newapi_param(
 
 
 def request_newsdataapi(
-    params: NewsDataApiParam, count: int
+    params: NewsDataApiParam, count: int, exact_count: bool
```
Comment (seanwu1105) [inline app.py:111]: Why do we need this? It seems to be set to `False` only.

**Review by seanwu1105 — APPROVED** [review]
Leaving some questions. If you see they are not problems, you can merge it directly.

File: app.py (hunk 2 above)
Comment (RaymondHuang210129) [reply app.py:111]: This is set to `True` when search_news() calls it.

File: app.py (hunk 1 above)
Comment (RaymondHuang210129) [reply app.py:305]: I am planning to refactor the filter part later. I am seeking some meta programming approaches to generalize the filter functions.

**Pull request merged**